spec/controllers/spree/products_controller_spec.rb

```diff
@@ -23,11 +22,10 @@
       product.stores << store_1
     end
 
-    it 'should return 404' do
+    it 'should raise ActiveRecord::RecordNotFound' do
+      skip if Spree.version.to_d < 3.3
```

Comment:
@VygovskySergey let's add a comment why we're skipping this for previous versions with a link to 404 spree PR